src/types/index.ts

```diff
@@ -185,3 +185,21 @@ export type ToastProps = {
    */
   onPress?: () => void;
 };
+
+export type TSafeArea = {
+  top: number
+  bottom: number
+  left: number
+  right: number
+}
+
+export type TNativeModulesSafeArea = {
+  SafeAreaInsetsModule: {
+    getSafeAreaInsets(p: (error: Error, result: TSafeArea) => void): Promise<{
+      top: number;
+      bottom: number;
+      left: number;
+      right: number
+    }>;
+  };
```

Comment:
small suggestion: better keep these safe area types local to the `useSafeArea` hook, as they are internal-only (not exposed to the users of the library).